src/internal/SetupResumeAcceptor.cpp

```diff
@@ -198,13 +206,42 @@ void SetupResumeAcceptor::closeAndRemoveConnection(
     const std::shared_ptr<FrameTransport>& transport,
     folly::exception_wrapper ex) {
   transport->close(ex);
-  connections_.erase(transport);
+  connections_->erase(transport);
 }
 
 void SetupResumeAcceptor::removeConnection(
     const std::shared_ptr<FrameTransport>& transport) {
   transport->setFrameProcessor(nullptr);
-  connections_.erase(transport);
+  connections_->erase(transport);
+}
+
+folly::Future<folly::Unit> SetupResumeAcceptor::close() {
+  folly::Promise<folly::Unit> closingPromise;
+  auto closingFuture = closingPromise.getFuture();
+
+  if(eventBase_->isInEventBaseThread()) {
+    closeAllConnections();
+    closingPromise.setValue();
```

Comment:
you could just do `return makeFuture<folly::Unit>()` here and move the Promise into the `else`